src/ProxyManager/ProxyGenerator/Util/Properties.php

```diff
@@ -64,6 +64,27 @@ public function filter(array $excludedProperties) : self
         return new self($properties);
     }
 
+    /**
+     * Properties that cannot be referenced are non-nullable typed properties that aren't initialised
+     */
+    public function withoutNonReferenceableProperties() : self
+    {
+        return new self(array_filter($this->getInstanceProperties(), function (ReflectionProperty $property) : bool {
+            if (! $property->hasType()) {
+                return true;
+            }
+
+            /** @var $type \ReflectionType */
+            $type = $property->getType();
+
+            if ($type->allowsNull()) {
+                return true;
+            }
+
+            return isset($property->getDeclaringClass()->getDefaultProperties()[$property->getName()]);
```

Comment:
```php
return $property->getType()->allowsNull()
    || isset($property->getDeclaringClass()->getDefaultProperties()[$property->getName()]);
```

Reply:
If there's no type, that will crash

Reply:
Isn't that the same as the current code? Note that `$type` needs to be checked for nullability

Reply:
We could do the unmaintainable version:

```php
return false === $property->hasType()
    || $property->getType()->allowsNull()
    || isset($property->getDeclaringClass()->getDefaultProperties()[$property->getName()]);
```

Reply:
That wouldn't pass static analysis :-)

Reply:
This has been tested and changed, since I had to spawn https://bugs.php.net/bug.php?id=77673